repose-aggregator/components/filters/saml-policy-translation-filter/src/test/scala/org/openrepose/filters/samlpolicy/SamlPolicyTranslationFilterTest.scala

```diff
@@ -114,7 +119,45 @@ class SamlPolicyTranslationFilterTest extends FunSpec with BeforeAndAfterEach wi
   }
 
   describe("translateResponse") {
-    pending
+    val documentString =
+      """
+        |<saml2p:Response xmlns:saml2p="urn:oasis:names:tc:SAML:2.0:protocol" xmlns:xs="http://www.w3.org/2001/XMLSchema"/>
+      """.stripMargin
+    val document = DocumentBuilderFactory.newInstance()
+      .newDocumentBuilder()
+      .parse(new InputSource(new StringReader(documentString)))
+    val brokenXslt =
+      """
+        |<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+        |                version="1.0">
+        |    <xsl:template match="/">
+        |        <xsl:message terminate="yes">Break ALL the things!</xsl:message>
+        |    </xsl:template>
+        |</xsl:stylesheet>
+      """.stripMargin
+    val brokenXsltExec = new Processor(false).newXsltCompiler()
+      .compile(new StreamSource(new StringReader(brokenXslt)))
+    val workingXslt =
+      """
+        |<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+        |                version="1.0">
+        |    <xsl:template match="/">
+        |        <xsl:copy-of select="."/>
+        |    </xsl:template>
+        |</xsl:stylesheet>
+      """.stripMargin
+    val workingXsltExec = new Processor(false).newXsltCompiler()
+      .compile(new StreamSource(new StringReader(workingXslt)))
+
+    it("should throw a SamlPolicyException(400) if the translation fails") {
+      intercept[SamlPolicyException] {
+        filter.translateResponse(document, brokenXsltExec)
+      }.statusCode shouldEqual SC_BAD_REQUEST
+    }
+
+    it("should return the translated document without throwing an exception") {
+      filter.translateResponse(document, workingXsltExec)
```

Comment:
Maybe check that you have different documents? Or better yet, have the xslt actually do something, and look for that in the resulting document.